Algorithmia/client.py

```diff
@@ -141,7 +145,7 @@ def postJsonHelper(self, url, input_object, parse_response_as_json=True, **query
 
         response = self.requestSession.post(self.apiAddress + url, data=input_json, headers=headers, params=query_parameters)
 
-        if parse_response_as_json:
+        if parse_response_as_json and response.status_code == 200:
```

Comment:
because of the fact that this postJsonHelper actually triggers the post operation, it seems like unless we want to make a dedicated response routing workflow, this is the simplest solution. Again, not elegant and there might be something better.